src/main/java/com/wire/bots/sdk/server/resources/MessageResourceBase.java

```diff
@@ -108,7 +109,11 @@ protected void handleMessage(Payload payload, WireClient client) throws Exceptio
             case "user.connection": {
                 Logger.debug("user.connection: bot: %s, status: %s", botId, payload.connection.status);
                 if (payload.connection.status.equals("pending")) {
-                    client.acceptConnection(payload.connection.to);
```

Comment:
Let's introduce: `handler.onConnectRequest(status, user)`

Reply:
Do you think the `handler.onConnectRequest` method should return a boolean to indicate whether the connection should be accepted or not?

Reply:
It has the client passed as parameter so it can accept/decline the req. in it's body. Maybe to return true/false just as indication what had happened.
The question is should this method be implemented to accept all `pending` requests and allow Developers to override it in case they want to implement some kind of white listing for people that can and cannot connect to the bot.
Or this method should be empty just like others and force all Devs to explicitly implement the logic in the child class. I am not sure what is better approach atm. Kind of inclined to first option (have the default logic in the parent class)

Reply:
I added a suggestion for the `handler.onConnectRequest` method in fb25ab876a3b02875494eccb9ef37ff65163b042. I agree that the default behavior should be to accept pending requests, with the option for developers to explicitly overwrite and change the behavior. This way also does not change the current behavior.

However, I'm still exposing the conversationId in the `acceptConnection` request, since no `conversation.create` event is dispatched and I haven't figured out a different way to query it, yet.

Reply:
Thanks. I just got confirmed that `conversation.create` event is sent. However, I will run my own tests to confirm this. `conversation.create` should arrive just after pending reqs has been accepted. It's possible that current code does not handle the json correctly and exception is thrown. Logger level DEBUG would tell us more.
You can leave `UUID client.acceptConnection`, it's correct thing to do to return the new convId